File: algorithms/Palindrome/Palindrome.cpp
#include <iostream>
#include <string>

using namespace std;


int main(){
	int num, len;
	string str, reversed;


	cout << "Enter a positive integer: ";
	cin >> num;

	str = to_string(num);
	reversed = str;

	len = reversed.length();
	for (int i = 0; i < len / 2; i++){
        swap(reversed[i], reversed[len - i - 1]); 
	}

	if (reversed == str) {
		cout << "The Number entered was a Palindrome" << endl;
	} else { 
		cout << "The Number entered was NOT a Palindrome" << endl; 
	}

	system("pause");
	return 0;
} 

File: algorithms/Palindrome/Palindrome.java
import java.util.Scanner;

public class Palindrome {
    private static Scanner scanner = new Scanner( System.in );

    public static void main(String args[]) {
        String str, reversed;
        byte[] strArr, reversedArr;

        System.out.println("Enter a positive integer: ");
        str = scanner.nextLine();

        strArr = str.getBytes(); 
        reversedArr = new byte[strArr.length];
        
        for (int i = 0; i < strArr.length; i++) {
            reversedArr[i] = strArr[strArr.length - i - 1];
        }
        reversed = new String(reversedArr);

        
        if (str.equalsIgnoreCase(reversed)) {
            System.out.println("The Number entered was a Palindrome");
        } else { 
            System.out.println("The Number entered was NOT a Palindrome");
        }
    }
}